front50-s3/src/main/java/com/netflix/spinnaker/front50/config/S3Properties.java

```diff
@@ -29,6 +30,11 @@
   @NestedConfigurationProperty
   S3EventingProperties eventing = new S3EventingProperties();
 
+  // Front50 retrieves objects in batches of this size. Some S3 compatible store enforce a maximum number of keys
+  @Value("${spinnaker.s3.maxKeys:10000}")
```

Comment:
Don't think you should need this `@Value` ... you're already in a `@ConfigurationProperties` class rooted at `spinnaker.s3`.

`Integer maxKeys = 10000`